coordinator/internal/transitengine/crypto_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package transitengine
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	r "math/rand"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/edgelesssys/contrast/internal/crypto"
+
+	"github.com/stretchr/testify/require"
+
+	b64 "encoding/base64"
+)
+
+func TestCryptoAPICyclic(t *testing.T) {
+	t.Run("encrypt-decrypt handler", func(t *testing.T) {
+		for range 10 {
+			var choosenPlaintext string
+			plaintextBytes, err := crypto.GenerateRandomBytes(1 + r.Intn(99))
+			require.NoError(t, err)
+			key, err := crypto.GenerateRandomBytes(16)
+			require.NoError(t, err)
+			choosenPlaintext = b64.StdEncoding.EncodeToString(plaintextBytes)
+			t.Run("cyclic handler function testing", func(t *testing.T) {
+				var ciphertext, receivedPlaintext string
+				t.Run("encryption request handling", func(t *testing.T) {
+					jsonBody, err := createReqBodyJSON("plaintext", choosenPlaintext)
+					require.NoError(t, err)
+					req := httptest.NewRequest(http.MethodPut, "/v1/transit/encrypt/autounseal", bytes.NewReader(jsonBody))
+					req.Header.Set("Content-Type", "application/json")
+
+					encRespBody := getResponseBody(t, handleEncrypt, *req, key)
+					data, ok := encRespBody["data"].(map[string]any)
+					require.True(t, ok)
+					ciphertext, ok = data["ciphertext"].(string)
+					require.True(t, ok)
+				})
+				t.Run("decryption request handling", func(t *testing.T) {
+					decryptReqBody, err := createReqBodyJSON("ciphertext", ciphertext)
+					require.NoError(t, err)
```

Comment:
Instead of using `require.NoError(t, err)` we call `require := require.New(t)` in the first lines of a new test and then call `require.NoError(err)`.